rclcpp/src/rclcpp/node_interfaces/node_topics.cpp

```diff
@@ -140,6 +162,20 @@ NodeTopics::add_subscription(
     callback_group->add_waitable(subscription_event);
   }
 
+  if (use_intra_process) {
+    // Get the intra process manager for this context.
+    auto context = node_base_->get_context();
+    auto ipm =
+      context->get_sub_context<rclcpp::intra_process_manager::IntraProcessManager>();
+
+    // Use the id to retrieve the subscription intra-process from the intra-process manager.
+    auto subscription_intra_process =
+      ipm->get_subscription_intra_process(subscription->get_intra_process_id());
+
+    // Add to the callback group to be notified about intra-process msgs.
+    callback_group->add_waitable(subscription_intra_process);
```

Comment:
This line causes applications running with Fast-RTPS to hang up after calling `rclcpp::shutdown()`.

On the other hand the following works fine
```
get_node_base_interface()->get_default_callback_group()->add_waitable(subscription_intra_process);
```

Reply:
Actually lately I'm not able to reproduce the error anymore. Now it works